From 026842cb6df889c07e712442725a53a76f8a00ea Mon Sep 17 00:00:00 2001
From: Anton Arapov <anton@redhat.com>
Date: Tue, 6 Mar 2012 10:50:47 +0100
Subject: [PATCH 05/53] plugin: kerneloops: fix the libcurl use ("Expect:"
 header)

The patch fixes following issue:

* Connected to kerneloops.descope.org (184.72.83.237) port 80 (#0)
* > POST /submitoops.php HTTP/1.1
* Host: kerneloops.descope.org
* Accept: */*
Content-Length: 2444
Expect: 100-continue
Content-Type: multipart/form-data;
boundary=----------------------------bd6b682457c1
< HTTP/1.1 417 Expectation Failed
< Date: Tue, 06 Mar 2012 08:47:12 GMT
< Content-Length: 453
< Connection: close
< Content-Type: text/html; charset=iso-8859-1
<
* Closing connection #0

Signed-off-by: Anton Arapov <anton@redhat.com>
Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/plugins/reporter-kerneloops.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/plugins/reporter-kerneloops.c b/src/plugins/reporter-kerneloops.c
index 2d296cf..81ab93e 100644
--- a/src/plugins/reporter-kerneloops.c
+++ b/src/plugins/reporter-kerneloops.c
@@ -51,12 +51,18 @@ static CURLcode http_post_to_kerneloops_site(const char *url, const char *oopsda
     CURL *handle;
     struct curl_httppost *post = NULL;
     struct curl_httppost *last = NULL;
+    struct curl_slist *headers = NULL;
 
     handle = curl_easy_init();
     if (!handle)
         error_msg_and_die("Can't create curl handle");
 
+    headers = curl_slist_append(headers, "Accept: */*");
+    headers = curl_slist_append(headers, "Expect:");
+
     curl_easy_setopt(handle, CURLOPT_URL, url);
+    curl_easy_setopt(handle, CURLOPT_FAILONERROR, 1L);
+    curl_easy_setopt(handle, CURLOPT_HTTPHEADER, headers);
 
     curl_formadd(&post, &last,
             CURLFORM_COPYNAME, "oopsdata",
@@ -73,6 +79,7 @@ static CURLcode http_post_to_kerneloops_site(const char *url, const char *oopsda
     ret = curl_easy_perform_with_proxy(handle, url);
 
     curl_formfree(post);
+    curl_slist_free_all(headers);
     curl_easy_cleanup(handle);
 
     return ret;
-- 
1.7.11.2

